src/main-process/utils.js

```diff
@@ -0,0 +1,20 @@
+const electron = require('electron');
+
+module.exports.checkBounds = function(bounds) {
+  // check if the browser window is offscreen
+  const display = electron.screen.getDisplayNearestPoint(bounds).workArea;
+
+  const onScreen = bounds.x >= display.x
+        && bounds.x + bounds.width <= display.x + display.width
+        && bounds.y >= display.y
+        && bounds.y + bounds.height <= display.y + display.height;
+
+  if(!onScreen) {
+    delete bounds.x;
```

Comment:
I think V8 does not like removing props like this.
It might be better to construct a new object instead.
Besides, this would avoid mutating the function argument, which makes it easier to reason about.